apps/wallet/src/ui/app/redux/slices/originbyte-nfts/index.ts

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    createAsyncThunk,
+    createEntityAdapter,
+    createSlice,
+} from '@reduxjs/toolkit';
+
+import { NftClient } from './NftClient';
+import { parseIpfsUrl } from '_hooks';
+
+import type { ArtNft } from './NftClient';
+import type { RootState } from '_redux/RootReducer';
+import type { AppThunkConfig } from '_store/thunk-extras';
+
+const objectsAdapter = createEntityAdapter<ArtNft>({
+    selectId: ({ id }) => id,
+    sortComparer: (a, b) => a.id.localeCompare(b.id),
+});
+
+const isArtNft = (nft: ArtNft | undefined): nft is ArtNft => !!nft;
+
+export const fetchArtNft = createAsyncThunk<ArtNft[], string, AppThunkConfig>(
```

Comment:
We're trying to move our data fetching and async state management to [TanStack Query](https://tanstack.com/query/v4), especially for new things we're adding. There are some other examples in this repo that I think you could use as a reference for updating this to use `useQuery` instead.